src/farbfeld.rs

```diff
@@ -188,3 +225,30 @@ impl<W: Write> ImageEncoder for FarbfeldEncoder<W> {
         self.encode(buf, width, height)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::farbfeld::FarbfeldDecoder;
+    use crate::ImageDecoderExt;
+    use std::io::Cursor;
+
+    #[test]
+    fn read_rect() {
+        static RECTANGLE_IN: &[u8] =  b"farbfeld\
+                                        \x00\x00\x00\x02\x00\x00\x00\x03\
+                                        \x00\x01\x00\x02\x00\x03\x00\x04\x00\x02\x00\x03\x00\x04\x00\x05\
+                                        \x00\x03\x00\x04\x00\x05\x00\x06\x00\x04\x00\x05\x00\x06\x00\x07\
+                                        \x00\x05\x00\x06\x00\x07\x00\x08\x00\x06\x00\x07\x00\x08\x00\x09";
+        static RECTANGLE_OUT: &[u8] = b"farbfeld\
+                                        \x00\x00\x00\x01\x00\x00\x00\x02\
+                                        \x00\x04\x00\x05\x00\x06\x00\x07\
+                                        \x00\x06\x00\x07\x00\x08\x00\x09";
+
+        let mut out_buf = [0u8, 64];
```

Comment:
You used a comma rather than a semicolon so this is creating a 2 element array rather than a 64 element array.

Reply:
Though we should probably put an assertion in our load_rect helper function to make a nicer error message...

Reply:
Specifically, the top of load_rect should have a line something like:

```rust
        assert_eq!(u64::try_from(buf.len()), Ok(total_bytes));
```

Reply:
Dang, you're right! That's what I get for writing these so late.

Added a check, looks like this when failing:
```
thread 'farbfeld::tests::read_rect' panicked at 'assertion failed: `(left == right)`
  left: `Ok(2)`,
 right: `Ok(16)`: output buffer too short', src\image.rs:227:5
```